File: 26.remove-duplicates-from-sorted-array.py
#
# @lc app=leetcode id=26 lang=python
#
# [26] Remove Duplicates from Sorted Array
#

# @lc code=start
class Solution(object):
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        elems = 1
        
        for i in range(1, len(nums)):
            if nums[i] != nums[i-1]:
                nums[elems] = nums[i]
                elems += 1
        return elems

        
# @lc code=end

File: solutions_java/arrays/26.remove-duplicates-from-sorted-array.java
/*
 * @lc app=leetcode id=26 lang=java
 *
 * [26] Remove Duplicates from Sorted Array
 */

// @lc code=start

import java.util.HashSet;
import java.util.Set;

class Solution {
    public int removeDuplicates(int[] nums) {
        Set<Integer> set = new HashSet<>();
        int start = 0;
        int end = nums.length-1;
        while (start <= end) {
            if (!set.contains(nums[start])) {
                set.add(nums[start]);
                start++;
            } else {
                while (set.contains(nums[end]) && start < end) {
                    end--;
                }
                nums[start] = nums[end];
                end--;
            }
        }
        Arrays.sort(nums, 0, set.size());
        return end + 1;
    }
}
// @lc code=end

